Zend/zend_ini.c

```diff
@@ -536,6 +538,75 @@ ZEND_API bool zend_ini_parse_bool(zend_string *str)
 	}
 }
 
+ZEND_API zend_long zend_ini_parse_quantity(zend_string *value, zend_string **errstr) /* {{{ */
+{
+	char *digits_end = NULL;
+	char *str = ZSTR_VAL(value);
+	size_t str_len = ZSTR_LEN(value);
+
+	/* Ignore trailing whitespace */
+	while (str_len && ZEND_IS_WHITESPACE(str[str_len-1])) --str_len;
+	if (!str_len) {
+		*errstr = NULL;
+		return 0;
+	}
+
+	/* Perform following multiplications on unsigned to avoid overflow UB.
+	 * For now overflow is silently ignored -- not clear what else can be
+	 * done here, especially as the final result of this function may be
+	 * used in an unsigned context (e.g. "memory_limit=3G", which overflows
+	 * zend_long on 32-bit, but not size_t). */
+	zend_ulong retval = (zend_ulong) ZEND_STRTOL(str, &digits_end, 0);
+
+	if (digits_end == str) {
+		*errstr = zend_strpprintf(0, "Invalid numeric string '%.*s': no valid leading digits, interpreting as '0' for backwards compatibility",
+						(int)str_len, str);
+		return 0;
+	}
+
+	/* Allow for whitespace between integer portion and any suffix character */
+	while (ZEND_IS_WHITESPACE(*digits_end)) ++digits_end;
+
+	/* No exponent suffix. */
+	if (!*digits_end) {
```

Comment:
Can `NUL` bytes be inserted via ini setting? If so, the rest of the string would be ignored. Not a major issue.

Reply:
Well spotted! This would have been a breaking change (minor, but still a breaking change)